hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java

```diff
@@ -45,6 +46,7 @@
   public HiveClientPool(int poolSize, Configuration conf) {
     super(poolSize, TTransportException.class);
     this.hiveConf = new HiveConf(conf, HiveClientPool.class);
+    this.hiveConf.addResource(conf);
```

Comment:
It is strange that we have to provide the `conf` twice in this case.
Do I understand correctly that this is because of the precedence? So in the constructor we set the `conf` properties, then initialize based on the class, but we would like to have the `conf` values higher priority than ones stored in the jars?
Would this work?
```
    this.hiveConf = new HiveConf(HiveClientPool.class);
    this.hiveConf.addResource(conf);
```

Sorry to forgot about this PR (had little bit too much in the air) 😢

Reply:
Hi peter, your code works fine under this test, but `new HiveConf(conf, HiveClientPool.class)` will copy all the `overlay` properties in `conf` and `new HiveConf(HiveClientPool.class)` will not. 
In this test these two properties will be record in `overlay` properties (have higher priority):
```java
conf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, "file:/mywarehouse/");
conf.setInt("iceberg.hive.client-pool-size", 10);
```
So i think we should keep using `new HiveConf(conf, HiveClientPool.class)`.

Reply:
> new HiveConf(conf, HiveClientPool.class) will copy all the overlay properties in conf and new HiveConf(HiveClientPool.class) will not.

I agree with your assessment above. My guess is that `this.hiveConf.addResource(conf);` will add the overlay properties too (might be wrong, just asking). If this statement is right we might not need to add it twice.

Thanks, Peter

Reply:
As i tested, `this.hiveConf.addResource(conf)` will not copy the overlay properties, so the test can still pass, but `this.hiveConf.overlay` will not contains the origin overlay properties in `conf`.